apps/hash-frontend/src/pages/[shortname]/entities/[entity-uuid].page/query-editor-page.tsx

```diff
@@ -0,0 +1,193 @@
+  const entity = getRoots(entityEditorProps.entitySubgraph)[0];
+  const defaultValue = (entity?.properties as any)[
+    QUERY_PROPERTY_TYPE_BASE_URL
+  ];
+
+  const handleQueryEntities = useCallback(
+    async (multiFilter: MultiFilter) => {
+      const res = await queryEntities({
+        data: {
+          operation: { multiFilter },
+          graphResolveDepths: zeroedGraphResolveDepths,
+        },
+      });
+
+      if (!res.data) {
+        throw new Error(res.errors?.[0]?.message ?? "Unknown error");
+      }
+
+      return getRoots(res.data);
+    },
+    [queryEntities],
+  );
+
+  return (
+    <>
+      <Head>
+        <title>{entityLabel} | Entity | HASH</title>
+      </Head>
+      <EntityPageWrapper
+        header={
+          <EntityPageHeader
+            entityLabel={entityLabel}
+            chip={
+              <OntologyChip
+                icon={<OntologyIcon />}
+                domain="hash.ai"
+                path={
+                  <Typography>
+                    <Typography
+                      color={(theme) => theme.palette.blue[70]}
+                      component="span"
+                      fontWeight="bold"
+                    >
+                      {owner}
+                    </Typography>
+                    <Typography
+                      color={(theme) => theme.palette.blue[70]}
+                      component="span"
+                    >
+                      /entities/
+                    </Typography>
+                    <Typography
+                      color={(theme) => theme.palette.blue[70]}
+                      component="span"
+                      fontWeight="bold"
+                    >
+                      {entityUuid}
+                    </Typography>
+                  </Typography>
+                }
+              />
+            }
+          />
+        }
+      >
+        <EntityEditorContextProvider {...entityEditorProps}>
+          <Box sx={{ display: "flex", flexDirection: "column", gap: 6.5 }}>
+            <TypesSection />
+
+            {loading ? (
+              <Box>Loading...</Box>
+            ) : (
+              <EntityQueryEditor
+                key={queryEditorKey}
+                readonly={entityEditorProps.readonly}
+                defaultValue={defaultValue}
+                entityTypes={entityTypes}
+                propertyTypes={propertyTypes}
+                queryEntities={handleQueryEntities}
+                onDiscard={() => {
+                  if (mode === "create") {
+                    return router.push("/new/entity");
+                  }
+
+                  /**
+                   * this is not the best way to do this, but it works for now
+                   * to discard changes, we just change the key to make `EntityQueryEditor` re-render,
+                   * which resets the state of `EntityQueryEditor`, and thus discards the changes
+                   */
+                  setQueryEditorKey((key) => key + 1);
```

Comment:
Since the query editor is controlling form state internally, could it clear its own state, and then call `onDiscard`?

Reply:
Yes, it can clear. Not sure if it's priority to spend time improving it's state management. Maybe we should keep the todo comment for now?